libs/api/src/main/java/de/cyface/api/ClassifiedSegmentRetriever.java

```diff
@@ -85,16 +119,50 @@ public Future<List<ClassifiedSegment>> loadSegments(final List<String> userIds,
             }
             query.put("latest_data_point", timeRestriction);
         }
+    }
 
-        final var find = dataClient.find(collectionName, query);
-        find.onSuccess(result -> {
-            try {
-                promise.complete(pojo(result));
-            } catch (RuntimeException e) {
-                promise.fail(e);
-            }
-        });
-        find.onFailure(promise::fail);
+    /**
+     * Constructs a fully initialized object of this object.
+     *
+     * @param collectionName The name of the collection in the Mongo database used to store deserialized measurements.
+     * @param segment The {@link ClassifiedSegment} to load all {@link ClassifiedMeasurementSegment}s for
+     */
+    public ClassifiedSegmentRetriever(final String collectionName, final ClassifiedSegment segment) {
```

Comment:
I added constructors for different queries:
- load all (measurement-based) segments by segment id (in this case a user-based segment, i.e. before we load all user-aggregated segments for the map-overview and when a user clicks on a specific user-aggregated segment, we load all measurement-aggregated segments for that segment ... i.e. the "history" of that segment

Reply:
Solving this with constructors is a strange choice in my opinion. I guess it works, but you loose type safety. For instance I can provide a ClassifiedSegmentRetriever to some other class, which is initialized with the wrong query, which will result in a runtime error, where I would actually expect a compile time error.